test/coreneuron/test_pointer.py

```diff
@@ -0,0 +1,316 @@
+    for i, cell in enumerate(m.cells):
+        pc.set_gid2node(i, pc.id())
+        sec = cell.secs[0]
+        pc.cell(i, h.NetCon(sec(0.5)._ref_v, None, sec=sec))
+
+    # "integrate" fabs(ia) in each axial_pp and use that as a source of spikes
+    # for actually testing that the checkpoint is working with POINTER.
+    # Would be much better if I knew how to get file mode coreneuron to
+    # print trajectories.
+    for i, p in enumerate(h.List("AxialPP")):
+        pc.set_gid2node(i + 100, pc.id())
+        pc.cell(i + 100, h.NetCon(p, None))
+
+    spktime = h.Vector()
+    spkgid = h.Vector()
+    pc.spike_record(-1, spktime, spkgid)
+    cvode.cache_efficient(1)
+    pc.set_maxstep(10)
+    h.finitialize(-65)
+    pc.nrncore_write("coredat")
+
+    # do a NEURON run to record spikes
+    def run(tstop):
+        pc.set_maxstep(10)
+        h.finitialize(-65)
+        pc.psolve(tstop)
+
+    run(10)
+    spikes_std = sortspikes(spktime, spkgid)
+
+    # Does it work in direct mode?
+    from neuron import coreneuron
+
+    coreneuron.enable = True
+    for perm in [0, 1]:
+        coreneuron.cell_permute = perm
+        run(5)
+        pc.psolve(10)
+        spikes = sortspikes(spktime, spkgid)
+        assert spikes_std == spikes
+    coreneuron.enable = False
+
+    # standard to compare with checkpoint series
+    tpnts = [5.0, 10.0]
+    for perm in [0, 1]:
+        print("\n\ncell_permute ", perm)
+        common = "-d coredat --voltage 1000 --verbose 0 --cell-permute %d" % (perm,)
+        # standard full run
+        runcn(common + " --tstop %g" % float(tpnts[-1]) + " -o coredat")
+        # sequence of checkpoints
+        for i, tpnt in enumerate(tpnts):
+            tend = tpnt
+            restore = " --restore coredat/chkpnt%d" % (i,) if i > 0 else ""
+            checkpoint = " --checkpoint coredat/chkpnt%d" % (i + 1,)
+            outpath = " -o coredat/chkpnt%d" % (i + 1,)
+            runcn(
+                common + " --tstop %g" % (float(tend),) + outpath + restore + checkpoint
+            )
+
+        # compare spikes
+        cmp_spks(
+            spikes_std, "coredat", ["coredat/chkpnt%d" % (i,) for i in range(1, 3)]
+        )
+
+    m._callback_setup = None
+    pc.gid_clear()
+    del m
+
+
+def cmp_spks(spikes, dir, chkpntdirs):
+    # sorted nrn standard spikes into dir/out.spk
+    f = open(dir + "/temp", "w")
+    for spike in spikes:
+        f.write("%.8g\t%d\n" % (spike[0], int(spike[1])))
+    f.close()
+    # sometimes roundoff to %.8g gives different sort.
+    srun("sortspike {}/temp {}/nrn.spk".format(dir, dir))
+
+    srun("sortspike {}/out.dat {}/out.spk".format(dir, dir))
+    srun("cmp {}/out.spk {}/nrn.spk".format(dir, dir))
+
+    cmd = "cat"
+    for i in chkpntdirs:
+        cmd = cmd + " " + i + "/out.dat"
+    srun(cmd + " > " + dir + "/temp")
+    srun("sortspike {}/temp {}/chkptout.spk".format(dir, dir))
+    srun("cmp {}/out.spk {}/chkptout.spk".format(dir, dir))
```

Comment:
just a comment: this all looks bash embedded inside python 😅 but I think we can leave this. In #1682 Olli refactor similar test inside CMake but this one doesn't include MPI or similar options. So hopefully we will be able to clean this after 8.1 by bringing coreneuron integrated into neuron.